cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,301 @@
+use std::{
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+// Loads project's figment and merges the bind cli arguments into it
+impl<'a> From<&'a BindArgs> for Figment {
+    fn from(args: &'a BindArgs) -> Self {
+        let root = match args.root.clone() {
+            Some(root) => root,
+            None => find_project_root_path().unwrap(),
+        };
+
+        Config::figment_with_root(root).merge(args)
+    }
+}
+
+impl<'a> From<&'a BindArgs> for Config {
+    fn from(args: &'a BindArgs) -> Self {
+        let figment: Figment = args.into();
+        Config::from_provider(figment).sanitized()
+    }
+}
+
+#[derive(Debug, Clone, Parser, Serialize)]
+pub struct BindArgs {
+    #[clap(
+        help = "The project's root path. By default, this is the root directory of the current Git repository or the current working directory if it is not part of a Git repository",
+        long,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub root: Option<PathBuf>,
+
+    #[clap(
+        help = "Path to where the contract artifacts are stored",
+        long = "out",
+        short,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(rename = "out", skip_serializing_if = "Option::is_none")]
+    pub out_path: Option<PathBuf>,
+
+    #[clap(
+        help = "A directory at which to write the generated rust crate",
+        long = "crate-root",
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub crate_root: Option<PathBuf>,
+
+    #[clap(
+        long = "crate-name",
+        help = "The name of the rust crate to generate. This should be a valid crates.io crate name. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_name: Option<String>,
+
+    #[clap(
+        long = "crate-version",
+        help = "The version of the rust crate to generate. This should be a standard semver version string. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_version: Option<String>,
+}
+
+impl BindArgs {
+    /// Get the path to the foundry artifacts directory
+    fn artifacts(&self) -> PathBuf {
+        let c: Config = self.into();
+        c.out
+    }
+
+    /// Get the path to the root of the autogenerated crate
+    pub fn crate_root(&self) -> PathBuf {
+        self.crate_root.clone().unwrap_or_else(|| {
+            let mut path = self.artifacts();
+            path.push("bindings");
+            path
+        })
+    }
+
+    /// Get the name of the autogenerated crate
+    pub fn crate_name(&self) -> String {
+        self.crate_name.clone().unwrap_or_else(|| "foundry-contracts".to_owned())
+    }
+
+    /// Get the version of the autogenerated crate
+    pub fn crate_version(&self) -> String {
+        self.crate_version.clone().unwrap_or_else(|| "0.0.1".to_owned())
+    }
+
+    pub fn crate_src_dir(&self) -> PathBuf {
+        let mut src_dir = self.crate_root();
+        src_dir.push("src");
```

Comment:
would this be the same as `.join("src")`?
so this and the other functions could be simplified as `self.crate_root().join("src")`?